pkg/nsx/services/vpc/vpc.go

```diff
@@ -558,6 +558,15 @@ func (s *VPCService) CreateOrUpdateVPC(obj *v1alpha1.NetworkInfo) (*model.Vpc, *
 		return existingVPC[0], &nc, nil
 	}
 
+	if nc.VPCConnectivityProfile != "" {
```

Comment:
No need to create private IPBlock, you can just add private CIDR into VPC API request

Reply:
Put this VPCConnectivityProfile logic into `buildNSXVPC`?